include/circt/Dialect/MSFT/MSFT.td

```diff
@@ -57,14 +57,26 @@ class MSFT_Attr<string name, list<Trait> traits = [],
   let mnemonic = ?;
 }
 
+def RootedInstancePath : MSFT_Attr<"RootedInstancePath"> {
```

Comment:
This looks similar to firrtl.NonLocalAnchor, which is being used to store paths.  The main difference is the NLA defines a symbol and instances along the path have an attr reference to that symbol so you can tell locally they have something non-local going on.

Reply:
firrtl NLA also includes symbol references to the modules the instance names exist in to simplify walking.